File: backend/app/utils/download_dataset.js
const https = require('https');
const fs = require('fs');
const path = require('path');

const FILE_ID = '1h0pt48eJAq-wAHUf5-gFA9PlibOxVyk0';
const RAW_DIR = path.join(__dirname, '..', '..', '..', 'data', 'raw');
fs.mkdirSync(RAW_DIR, { recursive: true });

const dest = path.join(RAW_DIR, 'dataset_downloaded.bin');

function download(url, destPath) {
  https.get(url, (res) => {
    if (res.statusCode === 302 || res.statusCode === 301 || res.statusCode === 303) {
      download(res.headers.location, destPath);
      return;
    }

    const file = fs.createWriteStream(destPath);
    res.pipe(file);

    file.on('finish', () => {
      file.close(() => {
        console.log(`Download finished! File size: ${fs.statSync(destPath).size} bytes`);
      });
    });
  }).on('error', (err) => {
    console.error('Error downloading:', err.message);
  });
}

const initialUrl = `https://docs.google.com/uc?export=download&id=${FILE_ID}`;
console.log(`Starting download from ${initialUrl}...`);
download(initialUrl, dest);

File: backend/app/utils/download_dataset.py
import requests
import os
import zipfile

FILE_ID = "1h0pt48eJAq-wAHUf5-gFA9PlibOxVyk0"
RAW_DIR = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(__file__))), "data", "raw")
os.makedirs(RAW_DIR, exist_ok=True)

def download_file_from_google_drive(file_id, destination):
    URL = "https://docs.google.com/uc?export=download"
    session = requests.Session()

    response = session.get(URL, params={'id': file_id, 'confirm': 't'}, stream=True)
    
    # Save response content
    with open(destination, "wb") as f:
        for chunk in response.iter_content(32768):
            if chunk:
                f.write(chunk)
                
    print(f"Downloaded file to {destination}, size: {os.path.getsize(destination)} bytes")

if __name__ == "__main__":
    dest = os.path.join(RAW_DIR, "dataset_downloaded.bin")
    download_file_from_google_drive(FILE_ID, dest)
